### PR TITLE
Payment init API updated

### PR DESCRIPTION
**Pull request by Wiezzel**
Resolves: #395

### CODE REVIEW

File: goth/runner/cli/yagna_payment_cmd.py [hunk 1]
```diff
@@ -29,47 +30,69 @@ class PaymentStatus:
...
+        )
+
 
 class YagnaPaymentMixin:
     """A mixin class that adds support for `<yagna-cmd> payment` commands."""
 
+    def payment_fund(self: CommandRunner) -> None:
+        """Run `<cmd> payment fund`."""
+        args = make_args("payment", "fund")
+        self.run_command(args)
```
Comment (azawlocki) [inline goth/runner/cli/yagna_payment_cmd.py:54]:
```suggestion
        self.run_command(*args)
```

**Review by azawlocki — CHANGES_REQUESTED** [review]

File: test/goth/runner/cli/test_yagna_payment_cmd.py [hunk 2]
```diff
@@ -1,47 +1,41 @@
...
 def test_payment_init(yagna_container):
     """Test basic usage of `payment init` command."""
 
     yagna = Cli(yagna_container).yagna
 
-    # The test fails if we call `payment init` too fast
-    time.sleep(3.0)
-    yagna.payment_init()
+    status: PaymentStatus = yagna.payment_init()
+    assert status
```
Comment (azawlocki) [inline test/goth/runner/cli/test_yagna_payment_cmd.py:13]: Why do we need type annotation here? What does `assert status` check, if `status` is a `PaymentStatus` instance?

File: test/goth/runner/cli/test_yagna_payment_cmd.py [hunk 3]
```diff
@@ -1,47 +1,41 @@
...
-@pytest.mark.skip(reason="Not sure what is the expected behaviour")
 def test_payment_init_provider_mode(yagna_container):
-    """Test `payment init -p`."""
+    """Test `payment init --receiver`."""
 
     yagna = Cli(yagna_container).yagna
 
-    yagna.payment_init(provider_mode=True)
+    status: PaymentStatus = yagna.payment_init(receiver_mode=True)
+    assert status
```
Comment (azawlocki) [inline test/goth/runner/cli/test_yagna_payment_cmd.py:22]: Why do we need type annotation here? What does assert status check, if status is a PaymentStatus instance?

File: test/goth/runner/cli/test_yagna_payment_cmd.py [hunk 4]
```diff
@@ -1,47 +1,41 @@
...
-@pytest.mark.skip(reason="Not sure what is the expected behaviour")
 def test_payment_init_requestor_mode(yagna_container):
-    """Test `payment init -r`."""
+    """Test `payment init --sender`."""
 
     yagna = Cli(yagna_container).yagna
 
-    yagna.payment_init(requestor_mode=True)
+    status: PaymentStatus = yagna.payment_init(sender_mode=True)
+    assert status
```
Comment (azawlocki) [inline test/goth/runner/cli/test_yagna_payment_cmd.py:31]: Why do we need type annotation here? What does assert status check, if status is a PaymentStatus instance?

**Review by azawlocki — APPROVED** [review]
I've left some nitpicking comments but you're good to go!

File: test/goth/runner/cli/test_yagna_payment_cmd.py (hunk 2 above)
Comment (zakaprov) [reply test/goth/runner/cli/test_yagna_payment_cmd.py:13]: Good catch, this is a leftover from an earlier revision, this is no longer valid.

File: test/goth/runner/cli/test_yagna_payment_cmd.py (hunk 3 above)
Comment (zakaprov) [reply test/goth/runner/cli/test_yagna_payment_cmd.py:22]: See above!

File: test/goth/runner/cli/test_yagna_payment_cmd.py (hunk 4 above)
Comment (zakaprov) [reply test/goth/runner/cli/test_yagna_payment_cmd.py:31]: See above!

**Comment by zakaprov** [comment]
CI integration test failure is unrelated to these changes, I'm going to merge this and resolve the market issues in a separate PR.

**Pull request merged**